service/users.js

```diff
@@ -19,7 +19,13 @@ class UserService {
     return userRepository.addUser(userWithHashedPassword);
   }
 
-  removeUser(userName) {
+  async removeUser(userName) {
+    const user = await userRepository.getUserByUserName(userName);
+
+    if (!user) {
+      throw new InexistentItem('Username does not exists.');
+    }
+
     userRepository.deleteUser(userName);
```

Comment:
Here as well. I dont think you are await-ing it in the repository so the promise wont be await-ed at all